view/frontend/web/js/view/cart/product-protection-simple-offer.js

```diff
@@ -74,7 +74,7 @@ define(['cartUtils', 'extendSdk', 'ExtendMagento', 'stringUtils'], function (
 
     const activeProductData = {
       referenceId: config[0].selectedProductSku,
-      price: config[0].selectedProductPrice * 100,
+      price: parseInt(config[0].selectedProductPrice * 100).toPrecision(10),
```

Comment:
I don't think this is the best approach. This will return a string value like `'189899.0000'` when the previous implementation likely returned a number.

I'd recommend following the pattern we use for our `currency` [package](https://github.com/helloextend/currency/blob/main/src/math.ts#L9C10-L9C78) which takes into account the weird [rounding logic](https://www.jacklmoore.com/notes/rounding-in-javascript/) in JS

```
Number(`${Math.round(Number(`${price * 100}e${2}`))}e-${2}`)
```

As an example:
```
1898.99 -> 189899
199.99 -> 19999
110 -> 11000
1234567.89 -> 123456789 // Particularly important if large numbers come into play...
```

Reply:
please note that i took the existing code from [[helloextend/integration/view/frontend/web/js/view/catalog/product/product-protection-offer.js](url)](https://github.com/helloextend/magento-extend-integration/blob/master/view/frontend/web/js/view/catalog/product/product-protection-offer.js)
line 100